src/types/list.rs

```diff
@@ -20,10 +20,7 @@ pyobject_native_type!(PyList, ffi::PyList_Type, ffi::PyList_Check);
 
 impl PyList {
     /// Construct a new list with the given elements.
-    pub fn new<'p, T, U>(
-        py: Python<'p>,
-        elements: impl IntoIterator<Item = T, IntoIter = U>,
-    ) -> &'p PyList
+    pub fn new<T, U>(py: Python<'_>, elements: impl IntoIterator<Item = T, IntoIter = U>) -> &PyList
```

Comment:
Is this a new grammar?
Great.

Reply:
Oh it was stabilized a year ago https://github.com/rust-lang/rust/pull/49458.

Reply:
The [edition guide](https://doc.rust-lang.org/stable/edition-guide/) and the official rust blog (esp. the release notes) are really good resources for that